In [1]:
import pandas as pd
city=pd.read_csv("C:/Users/abooo/Downloads/mecca data.csv")

In [2]:
city

year  avg_temp
0    1843     25.16
1    1844     19.05
2    1845     22.46
3    1846       NaN
4    1847       NaN
..    ...       ...
166  2009     26.90
167  2010     27.30
168  2011     26.57
169  2012     27.02
170  2013     27.57

[171 rows x 2 columns]

In [3]:
# to check the missing values
city.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 171 entries, 0 to 170
Data columns (total 2 columns):
 #   Column    Non-Null Count  Dtype  
---  ------    --------------  -----  
 0   year      171 non-null    int64  
 1   avg_temp  156 non-null    float64
dtypes: float64(1), int64(1)
memory usage: 2.8 KB


In [4]:
#(2)
city.isnull().sum(axis=0)  # sum of rows

year         0
avg_temp    15
dtype: int64

since the avg tem has 156, which indcate that there is a missing values 15 rows, we can fill it with median

In [5]:
# replace missing values with median
median = city["avg_temp"].median() 
city["avg_temp"].fillna(median, inplace=True)
city.isnull().sum(axis=0)


year        0
avg_temp    0
dtype: int64

In [8]:
city.to_csv('city_cleand.csv',index=False)
city


year  avg_temp
0    1843    25.160
1    1844    19.050
2    1845    22.460
3    1846    25.605
4    1847    25.605
..    ...       ...
166  2009    26.900
167  2010    27.300
168  2011    26.570
169  2012    27.020
170  2013    27.570

[171 rows x 2 columns]

In [9]:
city.isnull().sum(axis=0) 

year        0
avg_temp    0
dtype: int64